src/components/UriViewer.js

```diff
@@ -53,6 +62,76 @@ const getMaxDownloadCostNA = bytes => {
   return downloadPrice < 0.01 ? '< $0.01' : Utils.formatUSD(downloadPrice)
 }
 
+const PreviewContent = ({ uri, metadata, metadata: { bucket, name }, googleProject }) => {
+  const signal = useCancellation()
+  const [preview, setPreview] = useState()
+  const loadPreview = async () => {
+    try {
+      const res = await Ajax(signal).Buckets.getObjectPreview(bucket, name, googleProject, isImage(metadata))
+      if (isImage(metadata)) {
+        setPreview(URL.createObjectURL(await res.blob()))
+      } else {
+        setPreview(await res.text())
+      }
+    } catch (error) {
+      setPreview(null)
+    }
+  }
+  Utils.useOnMount(() => {
+    if (isGs(uri) && isFilePreviewable(metadata)) {
+      loadPreview()
+    }
+  })
+  return els.cell([
+    Utils.cond(
+      [isGs(uri) && isFilePreviewable(metadata), () => h(Fragment, [
+        els.label('Preview'),
+        Utils.cond(
+          [preview === null, () => 'Unable to load preview.'],
+          [preview === undefined, () => 'Loading preview...'],
+          [isImage(metadata), () => img({ src: preview, width: 400 })],
+          () => div({ style: styles.previewText }, [preview]),
+        )
+      ])],
+      [!isGs(uri), () => els.label(`DOS uri's can't be previewed`)],
+      [isImage(metadata), () => els.label('Image is too large to preview')],
+      () => els.label(`File can't be previewed.`)
+    )
+  ])
+}
+
+const DownloadButton = ({ metadata: { bucket, name, size } }) => {
+  const signal = useCancellation()
+  const [url, setUrl] = useState()
+  const getUrl = async () => {
+    try {
+      const { url } = await Ajax(signal).Martha.getSignedUrl({ bucket, object: name }) // FIXME add provider
```

Comment:
Note: In the case of a RP bucket, this call will succeed, but the returned link will currently fail when accessed. We'll need to do something to (optionally) provide the `userProject` so that it gets included in the resulting URL.

Reply:
Turns out we just add `&userProject=foo` to the resulting URL.